Cannot get terraform to build using 'make'

Hi, I'm attempting to build terraform from source using make. In fact here's a handy Dockerfile to completely demonstrate the failure:

```
FROM ubuntu:latest

RUN apt-get update && apt-get install -y git curl wget golang build-essential
RUN bash -c 'mkdir -p /opt/go/src/github.com/{hashicorp,mitchellh}'

ENV GOPATH=/opt/go
ENV PATH=/usr/local/bin:/usr/bin:/bin:/usr/local/sbin:/usr/sbin:/sbin:/opt/go/bin

RUN ls /opt/go/src/github.com/

RUN cd /opt/go/src/github.com/hashicorp \
    && git clone https://github.com/hashicorp/terraform \
    && cd terraform \
    && git checkout v0.6.16 \
    && make
```

This always fails with:

```
--- FAIL: TestRead_PathNoPermission (0.00s)
    read_test.go:82: Expected error, got none!
FAIL
FAIL    github.com/hashicorp/terraform/helper/pathorcontents    0.003s
```

Is this test broken? It also fails when checking out master instead of v0.6.16. Thanks

--- a/helper/pathorcontents/read_test.go
+++ b/helper/pathorcontents/read_test.go
@@ -4,6 +4,7 @@ import (
 	"io"
 	"io/ioutil"
 	"os"
+	"os/user"
 	"strings"
 	"testing"
 
@@ -63,6 +64,14 @@ func TestRead_TildePath(t *testing.T) {
 }
 
 func TestRead_PathNoPermission(t *testing.T) {
+	// This skip condition is intended to get this test out of the way of users
+	// who are building and testing Terraform from within a Linux-based Docker
+	// container, where it is common for processes to be running as effectively
+	// root within the container.
+	if u, err := user.Current(); err == nil && u.Uid == "0" {
+		t.Skip("This test is invalid when running as root, since root can read every file")
+	}
+
 	isPath := true
 	f, cleanup := testTempFile(t)
 	defer cleanup()

